File: bsep_backend/src/main/java/com/example/bsep_backend/pki/service/CAAssignmentService.java
package com.example.bsep_backend.pki.service;

import com.example.bsep_backend.domain.User;
import com.example.bsep_backend.pki.domain.Certificate;
import com.example.bsep_backend.pki.dto.CAAssignmentRequest;
import com.example.bsep_backend.pki.dto.CAAssignmentResponse;

import java.util.List;

public interface CAAssignmentService {
    CAAssignmentResponse assignCAToCaUser(CAAssignmentRequest request, User admin);

    void revokeCAAssignment(Long assignmentId, User admin);

    List<CAAssignmentResponse> getAllAssignments();

    List<CAAssignmentResponse> getAssignmentsByOrganization(String organization);

    List<Certificate> getAssignedCertificatesForUser(User caUser);

    boolean canUserUseCertificate(User caUser, String certificateSerialNumber);

    List<CAAssignmentResponse> getAssignmentsForUser(User caUser);

    List<Certificate> getCertificatesInUserChain(User caUser);
}

File: bsep-frontend/src/services/caAssignmentService.ts
import api from '../api/api';
import { CAAssignmentRequest, CAAssignmentResponse } from '../types/caAssignment';

class CAAssignmentService {
  async assignCAToCaUser(request: CAAssignmentRequest): Promise<CAAssignmentResponse> {
    const response = await api.post('/ca-assignments', request);
    return response.data;
  }

  async revokeCAAssignment(assignmentId: number): Promise<string> {
    const response = await api.delete(`/ca-assignments/${assignmentId}`);
    return response.data;
  }

  async getAllAssignments(): Promise<CAAssignmentResponse[]> {
    const response = await api.get('/ca-assignments');
    return response.data;
  }

  async getAssignmentsByOrganization(organization: string): Promise<CAAssignmentResponse[]> {
    const response = await api.get(`/ca-assignments/organization/${organization}`);
    return response.data;
  }

  async getMyAssignments(): Promise<CAAssignmentResponse[]> {
    const response = await api.get('/ca-assignments/my');
    return response.data;
  }
}

export const caAssignmentService = new CAAssignmentService();
